### PR TITLE
Nav Unification: Remove old sidebar (part 4)

### PR DESCRIPTION
**Pull request by mmtr**
Part of https://github.com/Automattic/wp-calypso/issues/47051.

#### Changes proposed in this Pull Request
- Removes the `is-nav-unification` class added to the `body` element as part of a bigger cleanup task which seeks to remove no longer used code after the Nav Unification launch.
- Combines the `my-sites/sidebar` and `my-sites/sidebar-unified` folders since we always use the same sidebar now in My sites.

<details>
<summary><strong>Context</strong></summary>

Last year, we introduced new components for the navigation sidebar as part of the Nav Unification project. However, we've been falling back to the old sidebar components in these scenarios:
- As an escape hatch using [the `?disable-nav-unification` query param](https://github.com/Automattic/wp-calypso/blob/c7e06d5e2fe1ca94643d16855209704c39452f30/client/state/selectors/is-nav-unification-enabled.js#L9-L12).
- For self-hosted Jetpack sites [not updated to Jetpack 9.8](https://github.com/Automattic/wp-calypso/blob/c7e06d5e2fe1ca94643d16855209704c39452f30/client/state/selectors/is-nav-unification-enabled.js#L14-L19).
- [In Jetpack Cloud](https://github.com/Automattic/wp-calypso/blob/c7e06d5e2fe1ca94643d16855209704c39452f30/client/my-sites/navigation/index.jsx#L24-L25).

We no longer need to support the first two scenarios, since the escape hatch was meant to be temporary and Calypso does not need to support lower Jetpack versions than 9.8 anymore.

As for the third scenario, Jetpack Cloud only needs a small subset of the old components.

That gives us a good opportunity to remove a lot of unused code which would simplify the overall Calypso codebase.
</details>

#### Testing instructions

- Use the live links below for both Calypso and Jetpack Cloud.
- Make sure there are no regressions in the sidebar.

### CODE REVIEW

File: client/my-sites/sidebar-unified/style.scss [hunk 1]
```diff
@@ -41,167 +41,196 @@ $font-size: rem( 14px );
 	}
 }
 
-.is-nav-unification {
-	// client/layout/style.scss
-	.layout__content {
-		min-height: 101vh; // Hack to give JS the chance to trigger the scroll event when the content is short. JS code: client/layout/utils.ts:36.
+// client/layout/style.scss
```
Comment (mmtr) [inline client/my-sites/sidebar-unified/style.scss:44]: Changes here look much more bigger than they really are – I just removed the `.is-nav-unification {` wrapper and outdented the inner selectors in 1 tab.

Comment (mmtr) [reply client/my-sites/sidebar-unified/style.scss:44]: They are way easier to review by using the unified diff view with whitespaces hidden:
Screen Shot 2022-03-25 at 11 04 35

**Comment by mmtr** [comment]
This should be ready for review now.

File: client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx [hunk 2]
```diff
@@ -51,7 +51,7 @@ export default ( { path } ) => {
 					} ) }
 					link={ settingsPath( siteSlug ) }
 					onNavigate={ onNavigate }
-					selected={ itemLinkMatches( [ settingsPath( siteSlug ) ], path ) }
+					selected={ itemLinkMatches( settingsPath( siteSlug ), path ) }
```
Comment (cpapazoglou) [inline client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx:54]: Something seems going wrong here. Clicking `Settings` doesn't work as expected
CleanShot 2022-03-29 at 20 30 50

Comment (cpapazoglou) [reply client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx:54]: Hmm, this should be due to using the new `itemLinkMatches` 

https://github.com/Automattic/wp-calypso/pull/61903/files#diff-787a7578ecbdea54c723e0c6aadf924d402a13dea5fe61f4297ebdd7d385d8faR14

**Review by cpapazoglou — DISMISSED** [review]
That's a big PR but looks good! There is one regression in the Jetpack menu by using the new `itemLinkMatches`

https://github.com/Automattic/wp-calypso/pull/61903/files#diff-787a7578ecbdea54c723e0c6aadf924d402a13dea5fe61f4297ebdd7d385d8faR14

Apart from this, it should be good to go!

File: client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx (hunk 2 above)
Comment (mmtr) [reply client/components/jetpack/sidebar/menu-items/jetpack-cloud.jsx:54]: Good catch! Should've been fixed in 907ac96.

**Pull request merged**